### PR TITLE
feat : added coupon functionality

### PR DESCRIPTION
**Pull request by rutvikpumak**
user will able to apply coupon as well now
add some item in cart and click on apply

### CODE REVIEW

**Review by md-amir-gauhar — APPROVED** [review]
Looks good to merge

File: src/pages/Cart/component/CouponModal.js [hunk 1]
```diff
@@ -0,0 +1,63 @@
+import React from "react";
+import { useState } from "react";
+import { useOrder } from "../../../context/order/orderContext";
+import "./CouponModal.css";
+
+const COUPONS = [
+  { couponName: "DIWALI_DHAMAKA", value: 50 },
+  { couponName: "NEW_USER", value: 10 },
```
Comment (znjs) [inline src/pages/Cart/component/CouponModal.js:8]: use constants instead of strings to avoid typs

File: src/reducer/OrderReducer.js [hunk 2]
```diff
@@ -0,0 +1,36 @@
...
+    case ACTION_TYPE.PRICE_DETAILS:
+      const { priceOnMRP, discountOnMRP, couponApplied } = action.payload;
+      console.log({
+        ...state,
+        priceDetails: {
+          ...state.priceDetails,
+          price: price,
+          discount: discount,
+        },
+      });
```
Comment (znjs) [inline src/reducer/OrderReducer.js]: can remove the `console.log`

**Review by znjs — APPROVED** [review]
Good naming conventions used
good file structure
semantic HTML used
Preview working as expected

**Review by ScoobyDubeyDo — APPROVED** [review]
Liked the file structure and the code quality, also the coupon codes are working as expected 👍

**Comment by znjs** [comment]
deploy-preview-9--pustaka-react netlify app_cart
image
cart item count going negative

**Pull request merged**